pkg/controller/grafanadashboard/dashboard_pipeline.go

```diff
@@ -160,6 +174,22 @@ func (r *DashboardPipelineImpl) loadDashboardFromURL() error {
 	return nil
 }
 
+// Try to obtain the dashboard json from a config map
+func (r *DashboardPipelineImpl) loadDashboardFromConfigMap() error {
+	ctx := context.Background()
+	objectKey := client.ObjectKey{Name: r.Dashboard.Spec.ConfigMapRef.Name, Namespace: r.Dashboard.Namespace}
+
+	var cm corev1.ConfigMap
+	err := (*r.Client).Get(ctx, objectKey, &cm)
```

Comment:
Why dereference `r.Client` here?

Reply:
So this was done due to me not understanding the difference between a struct pointer and an interface pointer. I changed the client so that it is not an interface pointer.

Plus now I know what the difference is!